lib/spack/spack/oci/oci.py

```diff
@@ -69,6 +69,39 @@ def with_query_param(url: str, param: str, value: str) -> str:
     )
 
 
+def list_tags(ref: ImageReference) -> List[str]:
+    """Retrieves the list of tags associated with an image
+
+    Fetches all tags from an image's tags_url, handling pagination of
+    results.
+
+    Args:
+        reg: The image reference.
+    Returns:
+        List of strings representing all tags associated with the image.
+
+    """
+    tags = []
+    fetch_url: Optional[str] = ref.tags_url()
+
+    while fetch_url:
+        request = Request(url=fetch_url)
+        response = spack.oci.opener.urlopen(request)
+        spack.oci.opener.ensure_status(request, response, 200)
+        tags.extend(json.load(response)["tags"])
+        fetch_url = None
+        link_header = spack.util.web.get_header(response.headers, "Link")
+        if link_header:
+            uri_reference = link_header.split(";")[0].strip()
+            parsed = urllib.parse.urlparse(uri_reference)
```

Comment:
This is not following RFC5988.

Reply:
Sorry what part specifically?  Should I ensure I found `<` and `>` enclosing the uri, and strip them off first?